File: app/src/main/java/com/ruzhan/rxrepository/source/remote/RemoteDataSourceImpl.java
package com.ruzhan.rxrepository.source.remote;

import com.ruzhan.rxrepository.model.UserModel;
import com.ruzhan.rxrepository.network.RemoteApi;
import com.ruzhan.rxrepository.network.RemoteClient;

import io.reactivex.Single;
import io.reactivex.schedulers.Schedulers;

/**
 * create ruzhan: 2018/6/21 10:10
 */
public class RemoteDataSourceImpl implements IRemoteDataSource {

    private RemoteApi api;

    public RemoteDataSourceImpl() {
        api = RemoteClient.get();
    }

    @Override
    public Single<UserModel> getRemoteUser() {
        return api.getRemoteUser().subscribeOn(Schedulers.io());
    }
}

File: app/src/main/java/com/ruzhan/rxrepository/source/remote/RemoteDataSourceImpl.kt
package com.ruzhan.rxrepository.source.remote

import com.ruzhan.rxrepository.model.UserModel
import com.ruzhan.rxrepository.network.RemoteClient
import io.reactivex.Single
import io.reactivex.schedulers.Schedulers


class RemoteDataSourceImpl : IRemoteDataSource {

    private val api = RemoteClient.get()

    override val remoteUser: Single<UserModel>
        get() = api.remoteUser.subscribeOn(Schedulers.io())
}
